src/Web3.elm

```diff
@@ -186,17 +190,133 @@ toChecksumAddress (Address address) =
         }
 
 
+toWei : EthUnit -> String -> Result String BigInt
+toWei unit amount =
+    -- check to make sure input string is formatted correctly, should never error in here.
+    if Regex.contains (Regex.regex "^\\d*\\.?\\d+$") amount then
+        let
+            decimalPoints =
+                decimalShift unit
 
-{-
-   TODO
-   fromWei : EthUnit -> BigInt -> String
+            formatMantissa =
+                String.slice 0 decimalPoints >> String.padRight decimalPoints '0'
 
-   toWei : EthUnit -> String -> Maybe BigInt
+            finalResult =
+                case (String.split "." amount) of
+                    [ a, b ] ->
+                        a ++ (formatMantissa b)
 
-   bigIntToWei : EthUnit -> BigInt -> BigInt
+                    [ a ] ->
+                        a ++ (formatMantissa "")
 
--}
--- CORE
+                    _ ->
+                        "ImpossibleError"
+        in
+            case (BigInt.fromString finalResult) of
+                Just result ->
+                    Ok result
+
+                Nothing ->
+                    Err "There was an error calculating the result. However, the fault is not yours; please report this bug on github."
+    else
+        Err "Malformed number string passed to `toWei` function."
+
+
+fromWei : EthUnit -> BigInt -> String
+fromWei unit amount =
+    let
+        decimalIndex =
+            decimalShift unit
+
+        -- There are under 10^27 wei in existance (so we safe for the next couple of malenium of mining).
```

Comment:
Probably next several millennia when block rewards are cut come PoS

Reply:
Haha, yeah! (will fix my spelling on that too :laughing:)